my-dsl.md

```diff
@@ -1,38 +1,53 @@
 # Language
-_What is the name of the language? Link the name to its webpage 
-(if appropriate)._
 
+For this assignment I'm looking at the Hive Query Language (HQL), which is described in detail [here](https://cwiki.apache.org/confluence/display/Hive/LanguageManual).
+
+HQL the interface to the [Hive](https://en.wikipedia.org/wiki/Apache_Hive) pseudo-database. It allows for users to specify the storage of large amounts of data and do analysis on that data. It inherited its syntax and most of its semantics from SQL, with the exception of features added to support larger datasets.
 
 # Domain
-_Describe the language's domain in five words._
+*Big Data Analysis* and Storage
 
 
 # Computational model
-_What is the underlying computational model of this language? To answer this 
-question, provide a high-level description (no more than 100 words) of the 
-computation that occurs when someone executes a program in this language._
 
+An HQL query is executed in the following steps:
+  1. The query is translated into one or more Hadoop MapReduce.
+  2. Those jobs are queued
+  3. They are executed by a Hadoop cluster
+
+One item of note is that sequential jobs transfer information by writing it to the cluster's distributed filesystem (so the next job can read the data out of the file system and use it for its computation).
 
 # DSL-ness
-_Fowler writes about a spectrum of languages, from general-purpose languages to 
-"purely" domain-specific. Where does the DSL you chose fall on this spectrum, 
-and why?_ 
 
+I would place HQL pretty far to the 'DSL' side of the spectrum for the following reasons:
+  - Its syntax is very declarative, and only provides functionality for data analysis
+  - It provides no looping constructs and very limited conditional constructs
+  - It includes a number of features that cater directly to the needs of big data
 
 # Internal or external?
-_Is the language implemented as an internal or external DSL? 
-Justify your answer._
 
+It is an external DSL - it exists in its own files and is parsed, translated, and executed by an external execution engine.
 
 # Host language
-_What language(s) was (were) used to implement the DSL?_
 
+The HQL execution engine is written in Java. This is largely because HQL queries are ultimately executed as Hadoop MapReduce jobs, and Hadoop is dominantly Java (for better or worse).
 
```

Comment:
Are there sources for this? It would be cool to link to the implementation or description of the implementation.